OMCompiler/Compiler/SimCode/SimCodeUtil.mo

```diff
@@ -7896,7 +7902,14 @@ protected
   Integer len;
 algorithm
   // extract the sim var
-  simVar := dlowvarToSimvar(dlowVar, SOME(inAliasVars), inVars);
+  simVar := dlowvarToSimvar(dlowVar, SOME(inAliasVars), inVars, crefList);
+
+  // check for the FMI dump attribute is set to Dump all the fmi attributes
+  // TO Do for remaining Fmi attributes
+  if Flags.isSet(Flags.DUMP_FORCE_FMI_ATTRIBUTES) then
+    //simVar.causality := NONE();
+    //simVar.variability := NONE();
+  end if;
```

Comment:
Some pseudo code..:

if **not** Flags.isSet(Flags.DUMP_FORCE_FMI_ATTRIBUTES) then
  if simVar.causality == defaultCausality(simVar) then
    simVar.causality := NONE();
  end if;
  [...]
end if;